src/context/SendTokenContext.tsx

```diff
@@ -38,34 +41,73 @@ export const SendTokenProvider: React.FC<{ children: ReactNode }> = ({
 
   const [txError, setTxError] = useState<string>('');
 
+  const { sendMessageToMainTab } = useEventEmitterContext();
   const { pushChainClient, executorAddress } = usePushChain();
 
+  const isOpenedInIframe = !!getAppParamValue();
+
   const handleSendTransaction = async () => {
 
     try {
       setSendingTransaction(true);
       setTxError('')
       const value = parseUnits((amount || '0').toString(), tokenSelected.decimals);
 
-
-      const receipt = await pushChainClient.universal.sendTransaction({
+      const payload: ExecuteParams = {
         to: receiverAddress as `0x${string}`,
         value: value,
         data: "0x",
-      });
+      }
+
+      if (isOpenedInIframe) {
+        sendMessageToMainTab({
+          type: WALLET_TO_APP_ACTION.PUSH_SEND_TRANSACTION,
+          data: { ...payload, value: value.toString() },
+        });
+
+        return;
+      }
+
+      const receipt = await pushChainClient.universal.sendTransaction(payload);
 
       if (receipt.hash) {
         setSendState("confirmation");
         setTxhash(receipt.hash);
       }
+      setSendingTransaction(false);
     } catch (error) {
       console.error("Error in sending transaction", error);
       setTxError(error.message)
-    } finally {
       setSendingTransaction(false);
     }
   };
 
+  useEffect(() => {
```

Comment:
Can we shift this useEffect to be used inside EventEmitterContext.tsx file's useEffect. So that all the events are in one place.